tests/pathfinding/test_capacity_updates.py

```diff
@@ -0,0 +1,506 @@
+            channel_identifier=ChannelID(0),
+            token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS_BYTES,
+        ),
+        updating_participant=addresses[0],
+        other_participant=addresses[2],
+        updating_nonce=Nonce(1),
+        other_nonce=Nonce(1),
+        updating_capacity=TokenAmount(90),
+        other_capacity=TokenAmount(110),
+        reveal_timeout=2,
+        signature='',
+    )
+
+    with pytest.raises(MessageException) as exinfo:
+        pathfinding_service_mocked_listeners.on_pfs_update(message6)
+    print(str(exinfo.value))
+    assert 'Other Participant of Capacity Update does not match' in str(exinfo.value)
+
+
+def test_pfs_rejects_capacity_update_with_wrong_nonces(
+        addresses: List[Address],
+        pathfinding_service_mocked_listeners: PathfindingService,
+):
+    pathfinding_service_mocked_listeners.chain_id = 1
+
+    token_network = TokenNetwork(
+        token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS,
+        token_address=DEFAULT_TOKEN_ADDRESS,
+    )
+
+    pathfinding_service_mocked_listeners.token_networks[
+        token_network.address] = token_network
+
+    token_network.handle_channel_opened_event(
+        channel_identifier=ChannelIdentifier(0),
+        participant1=addresses[0],
+        participant2=addresses[1],
+        settle_timeout=15,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[0],
+        total_deposit=100,
+    )
+
+    token_network.handle_channel_new_deposit_event(
+        channel_identifier=ChannelIdentifier(0),
+        receiver=addresses[1],
+        total_deposit=100,
+    )
+
+    # Check that the new channel has id == 0
+    assert token_network.channel_id_to_addresses[ChannelIdentifier(0)] == (
+        addresses[0],
+        addresses[1],
+    )
+
+    message7 = UpdatePFS(
+        canonical_identifier=CanonicalIdentifier(
+            chain_identifier=ChainID(1),
+            channel_identifier=ChannelID(0),
+            token_network_address=DEFAULT_TOKEN_NETWORK_ADDRESS_BYTES,
+        ),
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+        updating_nonce=Nonce(1),
+        other_nonce=Nonce(0),
+        updating_capacity=TokenAmount(90),
+        other_capacity=TokenAmount(110),
+        reveal_timeout=2,
+        signature='',
+    )
+
+    # Check first capacity update succeeded
+    pathfinding_service_mocked_listeners.on_pfs_update(message7)
+    view_to_partner, view_from_partner = token_network.get_channel_view_perspective(
+        channel_identifier=ChannelIdentifier(0),
+        updating_participant=addresses[0],
+        other_participant=addresses[1],
+    )
+    assert view_to_partner.capacity == 90
+    assert view_to_partner.update_nonce == 1
+    assert view_from_partner.capacity == 110
+    assert view_from_partner.update_nonce == 0
+
+    message8 = UpdatePFS(
```

Comment:
might make sense to create a factory function for the `UpdatePFS` messages to reduce duplication.

Reply:
done

Reply:
done